src/controllers/auth.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable consistent-return */
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+import model from '../models/index';
+
+exports.signup = async (req, res) => {
+  const {
+    username, email, confirmPassword,
+  } = req.body;
+
+  let { password } = req.body;
+
+  if (password !== confirmPassword) {
+    return res.status(401).json({ message: "passwords don't match" });
+  }
+
+  await model.User.findOne({
+    where: {
+      email,
+
+    },
+
+  }).then((user) => {
+    if (user) {
+      return res.status(403).json({
+        error: 'email is already taken Login!',
+      });
+    }
+  });
+
+  bcrypt.hash(password, 10, (err, hash) => {
+    password = hash;
+
+    model.User.create({
+      username,
+      email,
+      password,
+
+    }).then((user) => {
+      const { id } = user;
+      const token = jwt.sign({ id }, process.env.secret);
+      return res.json({
+        token,
+      });
+    })
+
+      .catch((error) => error);
```

Comment:
This error handler is not actually sending the error to the client.

How to generate an error: delete the `secret` variable from your .`env` file, test the signup route and see if the error message propagates to you.

Reply:
yeah it wasn't throwing any error if you delete secret from  `.env`
`.catch((error) => next(error));` adding next fixes that